### PR TITLE
[Refactor python-gvm API] Step 2: Detach Assets, TLS Certificates, Split up Assets into Hosts, OS

### PR DESCRIPTION
**Pull request by y0urself**
**What**:

* Detach TLS Certificates
* Detach and split Assets into Hosts and OperatingSystems
  * Remove AssetType - its obsolete now
* Use explicit calls for hosts and operating_systems

**Why**:

**How**:

**Checklist**:

- [x] Tests
- [x] [CHANGELOG](https://github.com/greenbone/python-gvm/blob/master/CHANGELOG.md) Entry
- [x] Documentation

### CODE REVIEW

File: gvm/protocols/gmpv208/entities/hosts.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        host.add_element("type", "host")  # ignored for gmp7, required for gmp8
+        host.add_element("name", name)
+
+        if comment:
+            host.add_element("comment", comment)
+
+        return self._send_xml_command(cmd)
+
+    def delete_host(
+        self, *, host_id: Optional[str] = None, report_id: Optional[str] = None
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/entities/hosts.py:55]: This api looks very bad. both arguments are optional but in reality one of them is required. I would either split deleting a host by report_id into another function or just remove that functionality.

File: gvm/protocols/gmpv208/entities/hosts.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            raise RequiredArgument(
+                function=self.get_host.__name__, argument='host_id'
+            )
+
+        cmd.set_attribute("asset_id", host_id)
+        cmd.set_attribute("type", "host")
+
+        return self._send_xml_command(cmd)
+
+    def modify_host(self, host_id: str, comment: Optional[str] = "") -> Any:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/entities/hosts.py:125]: shouldn't it be
```suggestion
    def modify_host(self, host_id: str, *, comment: Optional[str] = "") -> Any:
```

Comment (bjoernricks) [reply gvm/protocols/gmpv208/entities/hosts.py:125]: And I am even not sure about the default value `""`

File: gvm/protocols/gmpv208/entities/operating_systems.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+from gvm.utils import add_filter
+from gvm.xml import XmlCommand
+
+
+class OperatingSystemsMixin:
+    def delete_operating_system(
+        self,
+        *,
+        operating_system_id: Optional[str] = None,
+        report_id: Optional[str] = None,
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/entities/operating_systems.py:33]: same as hosts i think this api should be split.

File: gvm/protocols/gmpv208/entities/operating_systems.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                argument='operating_system_id',
+            )
+
+        cmd.set_attribute("asset_id", operating_system_id)
+        cmd.set_attribute("type", "os")
+
+        return self._send_xml_command(cmd)
+
+    def modify_operating_system(
+        self, operating_system_id: str, comment: Optional[str] = ""
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/entities/operating_systems.py:105]: same as hosts. shouldn't it be
```suggestion
        self, operating_system_id: str, *, comment: Optional[str] = ""
```

File: gvm/protocols/gmpv208/entities/tls_certificates.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+        details: Optional[bool] = None,
+    ) -> Any:
+        """Request a list of TLS certificates
+
+        Arguments:
+            filter: Filter term to use for the query
+            filter_id: UUID of an existing filter to use for the query
+            include_certificate_data: Whether to include the certificate data in
+                the response
+
```
Comment (bjoernricks) [inline gvm/protocols/gmpv208/entities/tls_certificates.py:125]: details argument is missing in the docstring

**Review by bjoernricks — CHANGES_REQUESTED** [review]

File: gvm/protocols/gmpv208/entities/hosts.py (hunk 1 above)
Comment (y0urself) [reply gvm/protocols/gmpv208/entities/hosts.py:55]: yeah probably a good idea. I didn't look into all the functions in detail.

**Review by bjoernricks — APPROVED** [review]

**Pull request merged**